packages/react-async/src/utilities.ts

```diff
@@ -0,0 +1,19 @@
+import {Import} from '@shopify/async';
+
+export function normalize(module: any) {
+  if (module == null) {
+    return null;
+  }
+
+  const value = 'default' in module ? module.default : module;
+  return value == null ? null : value;
+}
+
+export async function resolve<T>(load: () => Promise<Import<T>>): Promise<T> {
+  try {
+    const resolved = await load();
+    return normalize(resolved);
+  } catch (error) {
+    throw Error('error loading module');
```

Comment:
Should we bother with this or just let the original error through?

Reply:
ya I am fine with that too